File: MichiMark/Persistence/Repository/Settings/Trans/TransRepository.swift
import Foundation

protocol TransRepository: Sendable {
    /// 論理削除されていない Action を全件取得
    func fetchAll() async throws -> [TransDomain]

    /// 新規作成 or 上書き保存
    func save(_ trans: TransDomain) async throws

    /// 既存 Action の更新
    func update(_ trans: TransDomain) async throws
}

File: flutter/lib/repository/trans_repository.dart
import '../domain/master/trans/trans_domain.dart';

/// 交通手段の永続化インターフェース
///
/// 削除は論理削除（isDeleted フラグ）で行うため、deleteメソッドは持たない。
/// 新規作成・更新は save で一本化する（upsert）。
abstract interface class TransRepository {
  /// 論理削除されていない交通手段を全件取得
  Future<List<TransDomain>> fetchAll();

  /// 交通手段を保存（新規作成 / 上書き更新）
  Future<void> save(TransDomain trans);
}
